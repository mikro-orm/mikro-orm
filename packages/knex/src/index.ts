/**
 * @packageDocumentation
 * @module knex
 */
/* istanbul ignore file */
export * from './AbstractSqlConnection';
export * from './AbstractSqlDriver';
export * from './AbstractSqlPlatform';
export * from './MonkeyPatchable';
export * from './SqlEntityManager';
export * from './SqlEntityRepository';
export * from './query';
export * from './schema';
export * from './typings';
export { SqlEntityManager as EntityManager } from './SqlEntityManager';
export { SqlEntityRepository as EntityRepository } from './SqlEntityRepository';

/** @ignore */
export { Knex, knex } from 'knex';
export * from '@mikro-orm/core';
