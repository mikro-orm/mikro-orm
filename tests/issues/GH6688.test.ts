import { Entity, PrimaryKey, Property, MikroORM, t, Embeddable, Embedded, Opt, Enum, Utils, AbstractSqlDriver } from '@mikro-orm/knex';
import { PLATFORMS } from '../bootstrap';

enum EnumItems {
  A = 'a',
  B = 'b',
  C = 'c',
}

@Embeddable()
class Account {

  @Property()
  type!: string;

}

@Entity()
class A {

  @PrimaryKey()
  id!: number;

  @Property({ type: t.array, nullable: false, default: ['en'] })
  languages: Opt<string[]> = ['en'];

  @Embedded(() => Account, { array: true, default: [] })
  accounts: Opt<Account[]> = [];

  @Enum({ items: () => EnumItems, array: true, default: [] })
  items: Opt<EnumItems[]> = [];

}

const options = {
  sqlite: { dbName: ':memory:' },
  mysql: { dbName: 'default_array_values', port: 3308 },
  mariadb: { dbName: 'default_array_values', port: 3309 },
  postgresql: { dbName: 'default_array_values' },
  mssql: { dbName: 'default_array_values', password: 'Root.Root' },
};

describe.each(Utils.keys(options))('default array values [%s]', type => {
  let orm: MikroORM<AbstractSqlDriver>;

  beforeAll(async () => {
    orm = await MikroORM.init<AbstractSqlDriver>({
      entities: [A],
      driver: PLATFORMS[type],
      ...options[type],
    });

    await orm.schema.ensureDatabase();
  });

  afterAll(() => orm.close(true));

  afterEach(() => orm.schema.dropSchema());

  test('database default when using arrays', async () => {
    const sql = await orm.schema.getCreateSchemaSQL();
    expect(sql).toMatchSnapshot();
    await orm.schema.execute(sql);
  });
});
