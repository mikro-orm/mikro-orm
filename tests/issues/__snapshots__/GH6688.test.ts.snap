// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default array values [better-sqlite] database default when using arrays 1`] = `
"create table \`a\` (\`id\` integer not null primary key autoincrement, \`languages\` text not null default 'en', \`accounts\` json not null default '[]', \`items\` text not null default '');
"
`;

exports[`default array values [mariadb] database default when using arrays 1`] = `
"set names utf8mb4;

create table \`a\` (\`id\` int unsigned not null auto_increment primary key, \`languages\` text not null default ('en'), \`accounts\` json not null default ('[]'), \`items\` text not null default ('')) default character set utf8mb4 engine = InnoDB;
"
`;

exports[`default array values [mssql] database default when using arrays 1`] = `
"create table [a] ([id] int identity(1,1) not null primary key, [languages] text not null constraint [a_languages_default] default 'en', [accounts] nvarchar(max) not null constraint [a_accounts_default] default '[]', [items] text not null constraint [a_items_default] default '');
"
`;

exports[`default array values [mysql] database default when using arrays 1`] = `
"set names utf8mb4;

create table \`a\` (\`id\` int unsigned not null auto_increment primary key, \`languages\` text not null default ('en'), \`accounts\` json not null default ('[]'), \`items\` text not null default ('')) default character set utf8mb4 engine = InnoDB;
"
`;

exports[`default array values [postgresql] database default when using arrays 1`] = `
"set names 'utf8';

create table "a" ("id" serial primary key, "languages" text[] not null default '{en}', "accounts" jsonb not null default '[]', "items" text[] not null default '{}');
"
`;

exports[`default array values [sqlite] database default when using arrays 1`] = `
"create table \`a\` (\`id\` integer not null primary key autoincrement, \`languages\` text not null default 'en', \`accounts\` json not null default '[]', \`items\` text not null default '');
"
`;
